File: LeetCode/112pathSum/Solution.java
import java.util.*;
public class Solution {
    public boolean hasPathSum(TreeNode root, int sum) {
        if(root==null) return false;
        LinkedList<TreeNode> stack=new LinkedList<TreeNode>();
        stack.push(root);
        TreeNode cur=root;
        int curSum=0;        
        while(!stack.isEmpty()){
            cur=stack.pop();
            //System.out.println(cur.val);
            if(cur.left==null && cur.right==null && cur.val==sum){
                return true;
            }
            if(cur.left!=null){
                cur.left.val+=cur.val;
                stack.push(cur.left);
            }
            if(cur.right!=null){
                cur.right.val+=cur.val;
                stack.push(cur.right);
            }    
        }
        return false;
    }
}

File: LeetCode/112pathSum/Solution.c
#define FALSE 0
#define TRUE 1
struct TreeNode {
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
};

bool hasPathSum(struct TreeNode* root, int sum) {
 	if(root==NULL) return FALSE;
 	int sum-=root.val;
 	if(sum==0) return TRUE;
 	if(sum<0) return FALSE;
 	return hasPathSum(root->left,sum)||hasPathSum(root->right,sum);
}
